plugins/modules/swim_workflow_manager.py

```diff
@@ -1513,78 +1532,121 @@ def get_diff_tagging(self):
         Description:
             This function tags or untags a software image as a golden image in Cisco Catalyst Center based on the provided
             tagging details. The tagging action is determined by the value of the 'tagging' attribute
-            in the 'tagging_details' dictionary.If 'tagging' is True, the image is tagged as golden, and if 'tagging'
+            in the 'tagging_details' dictionary. If 'tagging' is True, the image is tagged as golden, and if 'tagging'
             is False, the golden tag is removed. The function sends the appropriate request to Cisco Catalyst Center and updates the
             task details in the 'result' dictionary. If the operation is successful, 'changed' is set to True.
         """
 
         tagging_details = self.want.get("tagging_details")
         tag_image_golden = tagging_details.get("tagging")
         image_name = self.get_image_name_from_id(self.have.get("tagging_image_id"))
+        device_role = tagging_details.get("device_role", "ALL")
```

Comment:
self.log("Parsed device roles: {0}".format(device_roles_set), "DEBUG")

Reply:
updated the code